### PR TITLE
Add scss reset, mixins and utils

### PR DESCRIPTION
**Pull request by synbot**
### Acceptance Criteria
- [x] Add reset
- [x] Add mixins
- [x] Add layout etc. utils
- [x] reorder file structure (scss and components)

### CODE REVIEW

**Comment by natestormer** [comment]
:star2: @canfie1d @kylecesmat you both probably want to look at this

File: scss/_settings.scss [hunk 1]
```diff
@@ -1,9 +1,8 @@
 //
 // SYNFRASTRUCTURE SETTINGS
 //
-
 // Comment to remove rem-calc() from your settings
-@import "functions";
+@import "foundation/functions";
```
Comment (canfie1d) [inline scss/_settings.scss]: Did we not kill foundation directories yet? Should we?

File: scss/util/_margin.scss [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+// generation functions
+////////////////////////
+
+// @vars
+// directional values
+$directions                : top right bottom left;
+$max-margin-spacing        : 30 !default;
+$margin-spacing-threshhold : 3 !default;
+$margin-spacing-buffer     : $max-margin-spacing * $margin-spacing-threshhold;
+
```
Comment (canfie1d) [inline scss/util/_margin.scss]: I propose additional manually set helpers here.
.l--m-0
.l--p-0
.l--m-top-0

File: scss/util/_mixins.scss [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+@mixin filter($filter-type, $filter-amount) {
+    filter : $filter-type+unquote('(#{$filter-amount})');
+}
+
+@mixin absolute-center {
+  position    : absolute !important;
+    top       : 50%;
+    left      : 50%;
+    transform : translate(-50%, -50%);
```
Comment (canfie1d) [inline scss/util/_mixins.scss]: this needs a `width : 100%` as well.

File: scss/util/_mixins.scss [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+@mixin normalize-list {
+    margin          : 0;
+    padding         : 0;
+    list-style-type : none;
```
Comment (canfie1d) [inline scss/util/_mixins.scss]: Propose that this is set by default. (if the reset doesn't already do it)

**Comment by kylecesmat** [comment]
:+1:

File: scss/util/_margin.scss (hunk 2 above)
Comment (natestormer) [inline scss/util/_margin.scss]: i updated the loop to include zero values for all helpers

File: scss/util/_mixins.scss (hunk 4 above)
Comment (natestormer) [inline scss/util/_mixins.scss]: all lists are normalized as part of our reset. maybe we can take this out?

**Pull request merged**